src/main/java/uk/ac/ebi/ampt2d/storage/FileSystemStorageService.java

```diff
@@ -0,0 +1,82 @@
+package uk.ac.ebi.ampt2d.storage;
+
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.UrlResource;
+import org.springframework.stereotype.Service;
+import org.springframework.util.FileSystemUtils;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@Service
+public class FileSystemStorageService implements StorageService {
+    private Logger logger = Logger.getLogger(FileSystemStorageService.class);
+
+    private final Path rootLocation;
+
+    @Autowired
+    public FileSystemStorageService(StorageProperties properties) {
+        this.rootLocation = Paths.get(properties.getLocation());
+    }
+
+    @Override
+    public Path store(MultipartFile file) {
+        Path archivePath;
+
+        try {
+            if (file.isEmpty()) {
+                throw new StorageException("Failed to store empty file " + file.getOriginalFilename());
+            }
+
+            archivePath = this.rootLocation.resolve(file.getOriginalFilename());
+            logger.debug("Archiving file "+ file.getOriginalFilename() +" into " + archivePath.toString());
+            Files.copy(file.getInputStream(), archivePath);
+
+            return archivePath;
+        } catch (IOException e) {
+            throw new StorageException("Failed to store file " + file.getOriginalFilename(), e);
+        }
+    }
+
+    @Override
+    public Path load(String filename) {
+        return rootLocation.resolve(filename);
+    }
+
+    @Override
+    public Resource loadAsResource(String filename) {
+        try {
+            Path file = load(filename);
+            Resource resource = new UrlResource(file.toUri());
+            if(resource.exists() || resource.isReadable()) {
+                return resource;
+            }
+            else {
+                throw new StorageFileNotFoundException("Could not read file: " + filename);
```

Comment:
Probably just to differentiate from the java.io one. The all folder is from [https://github.com/spring-guides/gs-uploading-files/tree/master/complete/src/main/java/hello/storage](url)